main.cpp

```diff
@@ -23,7 +28,22 @@ int main()
 
     // Create an instance of effect_player implementation with cpp api, pass path to location of
     // effects and client token
-    auto ep = bnb::oep::interfaces::effect_player::create({BNB_RESOURCES_FOLDER}, BNB_CLIENT_TOKEN);
+    std::vector<std::string> dirs;
+#if defined(__APPLE__)
+    // The BNB SDK framework includes BNB resources if not compiled with option to separate resources.
+    CFBundleRef bundle = CFBundleGetMainBundle();
+    CFURLRef bundleURL = CFBundleCopyBundleURL(bundle);
+    char path[PATH_MAX];
+    Boolean success = CFURLGetFileSystemRepresentation(bundleURL, TRUE, (UInt8 *)path, PATH_MAX);
+    assert(success);
+    CFRelease(bundleURL);
+
+    dirs.push_back(std::string(path) + "/Contents/Frameworks/BanubaEffectPlayer.framework/Resources/bnb-resources");
+    dirs.push_back(std::string(path) + "/Contents/Resources");
+#else
+    dirs.push_back(BNB_RESOURCES_FOLDER);
+#endif
+    auto ep = bnb::oep::interfaces::effect_player::create(dirs, BNB_CLIENT_TOKEN);
```

Comment:
I think better to create a new function in main.cpp file. 
For example: std::vector<std::string> get_resource_directories()
and create ep like:
auto ep = bnb::oep::interfaces::effect_player::create(get_resource_directories(), BNB_CLIENT_TOKEN);

in this case the main() func will be simple